src/nvim/extmark.c

```diff
@@ -117,9 +117,16 @@ static void extmark_setraw(buf_T *buf, uint64_t mark, int row, colnr_T col, bool
   MarkTreeIter itr[1] = { 0 };
   MTKey key = marktree_lookup(buf->b_marktree, mark, itr);
   if (key.pos.row < 0 || (key.pos.row == row && key.pos.col == col)) {
+    // Does this hold? If it doesn't, we should still revalidate.
+    assert(!invalid || !mt_invalid(key));
```

Comment:
`extmark_setraw()` seems to allow returning early when the mark doesn't need to be moved. I don't know if it can happen but maybe the mark may still be invalid at that point? Added an assert; could also just add the right condition to continue instead.